src/components/EditRow.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react'
+import Select from 'react-select'
+import 'react-select/dist/react-select.css'
+import '../css/EditRow.css'
+
+class EditRow extends Component {
+  constructor (props) {
+    super(props)
+    this.optionsFromDataTypes = this.optionsFromDataTypes.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+    this.handleSelectChange = this.handleSelectChange.bind(this)
+    this.handleSave = this.handleSave.bind(this)
+  }
+
+  optionsFromDataTypes () {
+    let options = []
+    Object.keys(this.props.dataTypes).map((dataType, idx) => {
+      return options.push({value: dataType, label: dataType})
+    })
+    return options
+  }
+
+  handleChange (event) {
+    const target = event.target
+    const value = target.type === 'checkbox' ? target.checked : target.value
+    let newState = Object.assign({}, this.props.details)
```

Comment:
`const newState = {...this.props.details}`

because objects - as well as arrays - in JS use an object pointer unlike a primitive value like an int or string, even though you are adding and kv pair to `newState` you can still define it as a const